t/Util.pm

```diff
@@ -525,6 +555,88 @@ sub spawn_h2_server {
     return $server;
 }
 
+# usage: see t/90h2olog.t
+package H2ologTracer {
+    use POSIX ":sys_wait_h";
+
+    sub new {
+        my ($class, $opts) = @_;
+        my $h2o_pid = $opts->{pid} or Carp::croak("Missing pid in the opts");
+        my $h2olog_args = $opts->{args} // [];
+        my $h2olog_prog = t::Util::bindir() . "/h2olog";
+
+        my $tempdir = File::Temp::tempdir(CLEANUP => 1);
+        my $output_file = "$tempdir/h2olog.jsonl";
+        my $stderr_file = "$tempdir/stderr.txt";
+
+        pipe my $rfh, my $wfh or die "pipe failed: $!";
+
+        my $tracer_pid = fork;
+        die "fork(2) failed: $!" unless defined $tracer_pid;
+        if ($tracer_pid == 0) {
+            # child process, spawn h2olog
+            exec qq{$h2olog_prog @{$h2olog_args} -d -p $h2o_pid -w '$output_file' 2>$stderr_file};
+            die "failed to spawn $h2olog_prog: $!";
+        }
+
+        # wait until h2olog and the trace log becomes ready
+        my $get_trace;
+        STARTUP: while (1) {
+            Time::HiRes::sleep(0.1);
+            if (open my $efh, "<", $stderr_file) {
+                for (my $i = 0; $i < 10; $i++) {
+                    sleep(1);
+                    seek $efh, 0, 0 or die "seek failed: $!";
+                    local $/;
+                    if (my $stderr_content = scalar <$efh>) {
+                        Test::More::diag("h2olog[$tracer_pid]: $stderr_content");
+                        last STARTUP;
+                    }
+                }
+
+                Carp::confess "h2olog[$tracer_pid] emits nothing"
+            }
+            Carp::confess "h2olog[$tracer_pid] failed to start"
+                if waitpid($tracer_pid, WNOHANG) != 0;
+        }
```

Comment:
Assuming that all we need to do is wait for some output to stderr, I'm not sure if we need this complex loop.

Can we simply do `my $tracer_pid = open my $errfh, "|-", "exec $h2olog_prog ..."` followed by `<$errfh>`? If h2olog emits something, then we can continue. If it returns nothing, that means that h2olog has died, therefore the test should also die.

Reply:
Thanks for the suggestion. That's a really simpler! Done: 13b561a4db3555d75bd989fdf9e769d89ca26bf2